### ISSUE TITLE
Asciidoctor adds stray semicolon after greater-than-sign at end of URL template

### ISSUE THREAD

**Issue by zeusttu** [asciidoctor/asciidoctor#3128]
If I type a URL (or rather a URL template) that ends with a parameter (which I indicate by surrounding it with `<>`), Asciidoctor renders a stray semicolon at the end of the URL. Putting the URL between backticks (to format it as code) does not change this. So the following snippet:

```
Asciidoctor renders a semicolon at the end of http://<host>:<port>

It will still do this when I format it as code: `http://<host>:<port>`
```

Will render to:

> Asciidoctor renders a semicolon at the end of [http://&lt;host&gt;:&lt;port&gt;]();
> 
> It will still do this when I format it as code: `http://<host>:<port>;`

If I look it the generated HTML, I find that this is because the `>` is translated to `&gt;`, and then the URL template is interpreted by Asciidoctor as a link, and somehow the `;` in `&gt;` is seen as not being part of the link:

```
<p>Asciidoctor renders a semicolon at the end of <a href="http://http://&lt;host&gt;:&lt;port&gt" class="bare">http://&lt;host&gt;:&lt;port&gt</a>;</p>
```

Possibly related to #1819 and #61. Asciidoctor version (installed on ArchLinux with Pacman):

```
Asciidoctor 1.5.8 [https://asciidoctor.org]
Runtime Environment (ruby 2.6.1p33 (2019-01-30 revision 66950) [x86_64-linux]) (lc:UTF-8 fs:UTF-8 in:UTF-8 ex:UTF-8)
```

**Comment by mojavelinux** [comment]
You're correct that the processor mangles this input, but I would argue that it shouldn't be treated as a URL in the first place. The proper way to escape a URL in Asciidoctor is to either use a backslash:

```
\http://<host>:<port>
```

or, the more robust approach, an inline passthrough:

```
+http://<host>:<port>+
```

**Comment by zeusttu** [comment]
Thanks! This sounds like an acceptable workaround :)
